objax/nn/layers.py

```diff
@@ -175,6 +176,11 @@ def __init__(self,
 
     def __call__(self, x: JaxArray) -> JaxArray:
         """Returns the results of applying the convolution to input x."""
+        assert x.shape[1]%self.nin == 0, \
+            'Attempting to convolve an input with {} input channels ' \
+            'when the convolution expects {} channels.' \
+            'For reference, self.w.shape={} and x.shape={}.'.format(x.shape[1], self.nin, 
+                                                                    self.w.value.shape, x.shape)
```

Comment:
Please use f-strings and proper spacing and also I believe the check should be `a == b` rather than `a % b == 0`, also you refer to `self.w.shape`, I assume you really meant `self.w.value.shape`:
```suggestion
        assert x.shape[1] == self.nin, f'Attempting to convolve an input with {x.shape[1]} input channels ' \
                                   f'when the convolution expects {self.nin} channels.' \
                                   f'For reference, self.w.value.shape={self.w.value.shape} and x.shape={x.shape}.'
```